### ISSUE TITLE
[steamshovel] Duplicated screenshot implementation (Trac #2043)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2035]
The python api provides a way to take a screenshot, but this is out of sync with the menu-based screenshot.

Specifically for IceTop, the menu-based screenshot captures all the IceTop widgets while the python api version only does the main opengl window.

Migrated from https://code.icecube.wisc.edu/ticket/2043
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:14:03", 
    "description": "The python api provides a way to take a screenshot, but this is out of sync with the menu-based screenshot.\n\nSpecifically for IceTop, the menu-based screenshot captures all the IceTop widgets while the python api version only does the main opengl window.", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067243750755", 
    "component": "combo core", 
    "summary": "[steamshovel] Duplicated screenshot implementation", 
    "priority": "major", 
    "keywords": "steamshovel", 
    "time": "2017-06-23T20:37:41", 
    "milestone": "", 
    "owner": "sander.vanheule", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170626T12:32:30: 
* sander.vanheule changed owner from "" to "sander.vanheule"
* sander.vanheule commented:

> The same functionality is implemented twice in steamshovel, which is obviously A Bad Thing™. I took the liberty of changing the ticket description to match this view.
> 
> The screenshot code should probably be implemented in C++ (could just move the existing implementation), so it remains accessible to the screenshot dialog, and then be exposed via py-bindings to the scripting interface.
> 
> 


* sander.vanheule changed status from "new" to "accepted"
* sander.vanheule changed keywords from "" to "steamshovel"
* sander.vanheule changed summary from "[steamshovel] python api screenshot vs menu screenshot" to "[steamshovel] Duplicated screenshot implementation"

**Comment by icecube-trac** [comment]
Trac update at 20170706T12:39:07: 
* sander.vanheule commented:

> In changeset 156951:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="156951"
> Rework screenshot and 3D projection interface.
> 
> The single screenshot dialog is now split up into two actions:
> * Screenshot: captures the active view (3D, IceTopViewer,...)
>   Available in python via window.captureActiveView().
>   Available in the GUI via Window > Screenshot...
> * Projection: takes (composite) screenshots of the 3D view
>   Available in python via icecube.steamshovel.util.projection.get_projection()
>   Available in the GUI via Window > 3D Projection...
> 
> Fixes #2035. The old icecube.steamshovel.util.projection.screenshot() function
> is now replaced by get_projection(), while the new
> icecube.steamshovel.util.projection.screenshot function just refers to
> window.captureActiveView().
> ```


* sander.vanheule changed resolution from "" to "fixed"
* sander.vanheule changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:14:03: 
* olivas commented:

> Milestone August 2017 deleted


* olivas changed milestone from "August 2017" to ""
